src/folio-runner.js

```diff
@@ -0,0 +1,196 @@
+  }
+
+  const projectPath = runCfg.playwright.projectPath ? path.join(cwd, runCfg.playwright.projectPath) : cwd;
+  if (!fs.existsSync(projectPath)) {
+    throw new Error(`Could not find projectPath directory: '${projectPath}'`);
+  }
+
+  const folioProc = spawn(nodeBin, procArgs, {stdio: 'inherit', cwd: projectPath, env});
+
+  const folioPromise = new Promise((resolve) => {
+    folioProc.on('close', (code /*, ...args*/) => {
+      const hasPassed = code === 0;
+      resolve(hasPassed);
+    });
+  });
+
+  const api = new SauceLabs({
+    user: process.env.SAUCE_USERNAME,
+    key: process.env.SAUCE_ACCESS_KEY,
+    region
+  });
+
+  async function createJob (hasPassed, startTime, endTime, args, playwright) {
+    let tags = process.env.SAUCE_TAGS;
+    if (tags) {
+      tags = tags.split(',');
+    }
+    let sessionId;
+    if (process.env.ENABLE_DATA_STORE) {
+      // TODO: When we enable this make sure it's getting the proper parameters
+      sessionId = await createJobShell(tags, api);
+    } else {
+      sessionId = await createJobWorkaround(tags, api, hasPassed, startTime, endTime, args, playwright);
+    }
+    const containerLogFiles = LOG_FILES.filter(
+      (path) => fs.existsSync(path));
+
+    // Take the 1st webm video we find and translate it video.mp4
+    // TODO: We need to translate all .webm to .mp4 and combine them into one video.mp4
+    const webmFiles = glob.sync(path.join(cwd, '__assets__', '**', '*.webm'));
+    let videoLocation;
+    if (webmFiles.length > 0) {
+      videoLocation = path.join(cwd, '__assets__', 'video.mp4');
+      try {
+        await exec(`ffmpeg -i ${webmFiles[0]} ${videoLocation}`, {suppressLogs: true});
+      } catch (e) {
+        videoLocation = null;
+        console.error(`Failed to convert ${webmFiles[0]} to mp4: '${e}'`);
+      }
+    }
+
+    let files = [
+      path.join(cwd, 'console.log'),
+      path.join(cwd, '__assets__', 'junit.xml'), // TOOD: Should add junit.xml.json as well
+      ...containerLogFiles
+    ];
+
+    if (videoLocation) {
+      files.push(videoLocation);
+    }
+
+    await Promise.all([
+      api.uploadJobAssets(sessionId, {files})
+        .then(
+          (resp) => {
+            if (resp.errors) {
+              for (let err of resp.errors) {
+                console.error(err);
+              }
+            }
+          },
+          (e) => console.error('upload failed:', e.stack)
+        ),
+      api.updateJob(process.env.SAUCE_USERNAME, sessionId, {
+        name: jobName,
+        passed: hasPassed
+      })
+    ]);
+
+    return sessionId;
+  }
+
+  try {
+    let startTime = new Date().toISOString();
+    const hasPassed = await folioPromise;
+    let endTime = new Date().toISOString();
+
+    // If it's not a VM, than create the job and upload the assets
+    if (!process.env.SAUCE_VM) {
+      let sessionId = await createJob(hasPassed, startTime, endTime, args, runCfg.playwright);
+      let domain;
+      switch (region) {
```

Comment:
Would be good to have a handling for staging! Check cypress runner for how it does the TLD (.net vs .com) for staging :)